src/components/Textfield/Textfield.tsx

```diff
@@ -163,7 +175,8 @@ export default function TextField(props: Props): JSX.Element {
             {iconLeft && <Icon name={iconLeft} className='textfield-value--icon-left' />}
             <input
               autoFocus={autoFocus}
-              value={type === 'number' ? value : value || ''}
+              defaultValue={defaultValue}
+              value={type === 'number' || defaultValue !== undefined ? value : value || ''}
```

Comment:
I don't get why `defaultValue !== undefined` condition is added here

Reply:
When `defaultValue` is set, `value` should be `undefined`. Previously, `value` defaulted to an empty string (`''`) when not set. The new `defaultValue !== undefined` check ensures `value` is passed correctly when `defaultValue` is used.